facturacion_mexico/tests/test_pdf_custom_section.py

```
@@ -0,0 +1,198 @@
+		result = render_pdf_note_template(
+			"Pagar a {company} ${total} antes de {due_date}",
+			company="ACME",
+			total="$1,000.00",
+			due_date="31/07/2026",
+		)
+		self.assertIn("ACME", result)
+		self.assertIn("31/07/2026", result)
+
+	def test_variable_desconocida_lanza_error(self):
+		# Caso 11: placeholder inválido
+		with self.assertRaises(frappe.ValidationError):
```

Ruff (0.15.17)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)